PHP/PHP Source.sublime-syntax

```diff
@@ -541,6 +544,24 @@ contexts:
               pop: true
         - match: ','
           scope: punctuation.separator.php
+    # https://wiki.php.net/rfc/typed_properties_v2
+    - match: (?=\b(?:public|private|protected|static)\b)
+      push:
+        - match: (?=\bfunction\b)
+          pop: true
+        - match: \b(?:public|private|protected|static)\b
+          scope: storage.modifier.php
+        - match: (?=(?:\?\s*)?{{identifier}})
```

Comment:
Why push into a separate context? You could as well assing the scopes in this match and pop, since you're not doing anything meta.

Reply:
> Why push into a separate context?

Because I want it to do a deterministic parsing. For example,

```php
public MY_TYPE $name = '......';
public  ?MY_TYPE $name = '......';

// I want to assign a scope to "?" and especially "MY_TYPE"
```

If I don't going to a new context, I thought there is no way to know "MY_TYPE" is a constant or a class?

```php
public const MY_CONST = '...'; // public is can be omitted
```

Reply:
Caught a bug that `(?=\bfunction\b)` should be `(?=\b(?:const|function)\b)`.

Reply:
I specifically meant the context that is set through this match on line 554 where you already know to assign `support.class.php` because that's what you do later inside the context with only two patterns.

Reply:
Ah. I guess I now understand your word. Awesome catch!
So now it's like https://github.com/sublimehq/Packages/pull/1961/commits/443c71e9247705f01e5fe10862b8f1cc4828c3c2 .